fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -1373,24 +1493,24 @@ protected RecordQueryPlan addTypeFilterIfNeeded(@Nonnull CandidateScan candidate
     }
 
     @Nullable
-    private ScoredPlan planVersion(@Nonnull CandidateScan candidateScan,
-                                   @Nonnull KeyExpression indexExpr,
-                                   @Nonnull QueryRecordFunctionWithComparison filter,
-                                   @Nullable KeyExpression sort) {
+    private ScoredMatch planVersion(@Nonnull CandidateScan candidateScan,
+                                    @Nonnull KeyExpression indexExpr,
+                                    @Nonnull QueryRecordFunctionWithComparison filter,
+                                    @Nullable KeyExpression sort) {
         if (indexExpr instanceof VersionKeyExpression) {
-            final Comparisons.Comparison comparison = filter.getComparison();
-            final ScanComparisons comparisons = ScanComparisons.from(comparison);
             if (sort == null || sort.equals(VersionKeyExpression.VERSION)) {
-                IndexScanParameters scanParameters = IndexScanComparisons.byValue(comparisons);
-                RecordQueryPlan plan = new RecordQueryIndexPlan(candidateScan.index.getName(), scanParameters, candidateScan.reverse);
-                return new ScoredPlan(1, plan, Collections.emptyList(), false);
+                final Comparisons.Comparison comparison = filter.getComparison();
+                final ComparisonRanges comparisonRanges = ComparisonRanges.tryFrom(comparison);
+                return new ScoredMatch(1,
```

Comment:
Note to reviewer: This match sets `createsDuplicates` and `isStrictlySorted` to `false` which I think is same behavior as before.